sycl/source/device.cpp

```diff
@@ -134,13 +137,77 @@ bool device::has_extension(const std::string &extension_name) const {
   return impl->has_extension(extension_name);
 }
 
+#ifdef __INTEL_PREVIEW_BREAKING_CHANGES
+template <typename Param>
+typename detail::is_device_info_desc<Param>::return_type
+device::get_info_internal() const {
+  return impl->template get_info<Param>();
+}
+
+detail::string device::get_device_info(detail::string_view Type) const {
+  std::string Info;
+  if (Type == typeid(info::device::name).name()) {
+    Info = impl->template get_info<info::device::name>();
+  } else if (Type == typeid(info::device::vendor).name()) {
+    Info = impl->template get_info<info::device::vendor>();
+  } else if (Type == typeid(info::device::driver_version).name()) {
+    Info = impl->template get_info<info::device::driver_version>();
+  } else if (Type == typeid(info::device::version).name()) {
+    Info = impl->template get_info<info::device::version>();
+  } else if (Type == typeid(info::device::profile).name()) {
+    Info = impl->template get_info<info::device::profile>();
+  } else if (Type == typeid(info::device::opencl_c_version).name()) {
+    Info = impl->template get_info<info::device::opencl_c_version>();
+  } else if (Type == typeid(info::device::ext_intel_pci_address).name()) {
+    Info = impl->template get_info<info::device::ext_intel_pci_address>();
+  } else {
```

Comment:
https://en.cppreference.com/w/cpp/types/type_info:

> name | implementation defined name of the type

How can this work if user's host compiler uses different string than the compiler we used to build RT?

Reply:
The header file I converted to `detail::string`, so as long as the user's host compiler uses our header file, it can't use a different `detail::strin`g.

Reply:
@aelovikov-intel 
I used enum class instead of typeid().name() string.